### PR TITLE
Search: Add better upsell page

### LINKED ISSUE #23570: S2.8 Add better upsell page for user to purchase a plan
When a site is connected but site has no search plan, the dashboard currently shows grayed out toggles and a small nudge for instant search purchase. We'll need another better screen for user to purchase a plan.

Image from iOS

### PR DESCRIPTION
**Pull request by kangzj**
Fixes #23570 

#### Changes proposed in this Pull Request:
The PR proposes a upsell page when site is connected but has no search subscription. The page has so much in common with the connection page, and will combine them in the future.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Navigate to a site fully connected but has no search subscription
- Open `/wp-admin/admin.php?page=jetpack-search`
- Ensure the upsell page is displayed
- Ensure you could purchase search products thru the link and then return to Search Dashboard

image

### CODE REVIEW

**Review by jsnmoon — COMMENTED** [review]
Changes look good and work as expected. 

I have commandeered this PR and pushed up some commits for added polish. Could I get some additional feedback/testing, @Automattic/jetpack-search?

File: projects/packages/search/src/dashboard/components/dashboard/connection-page.scss [hunk 1]
```diff
@@ -1,6 +1,16 @@
 @import '@automattic/jetpack-base-styles/style';
 
 .jp-search-dashboard-connection-screen {
+	background: var(--jp-white-off);
+	align-items: center;
+	display: flex;
+	width: 100%;
+
+	// Targets AdminSectionHero component
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/dashboard/connection-page.scss:9]: Is there a class name we could target instead?

File: projects/packages/search/src/dashboard/components/dashboard/search-promotion.scss [hunk 2]
```diff
@@ -0,0 +1,16 @@
+.jp-search-dashboard-promotion {
+	h1 {
+		font-size: var( --font-title-large );
+		font-weight: 600;
+		line-height: 1.2;
+	}
+
+	ul.jp-product-promote li {
+		background: url( data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADAAAAAwCAMAAABg3Am1AAAANlBMVEVHcEwFnwUInggGnggGnggHnAcAnwUFnQcAnwcGnwkFnQgGnQgFnwcGnQYFnQcFnAcGnQkDnwdhiL0pAAAAEnRSTlMAMF//f2Aw7yBQ3+9gcIBgcED+HDbkAAAAZklEQVR4Ae3LNwICARDDQC0+cv7/Y8mwV9odSfWIcf/+VegnGkIvDaGXKvTTn/Gz+Uf5xTL0K1XotS7fs5H6GHvvaO8d7c3j7rdgHne/A/PYt/cO+R42oYdN6OEQetiFHo4A//6dAXqtBEkmtWutAAAAAElFTkSuQmCC )
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/dashboard/search-promotion.scss:9]: Can we add a comment saying what this image is?

File: projects/packages/search/src/dashboard/components/dashboard/wrapped-dashboard.jsx [hunk 3]
```diff
@@ -9,22 +9,26 @@ import { useSelect, select as syncSelect } from '@wordpress/data';
...
 import useConnection from './use-connection';
-import SearchDashboard from './index';
 import { STORE_ID } from 'store';
+import UpsellPage from './upsell-page';
+import SearchConnectionPage from './connection-page';
+import SearchDashboard from './index';
 
 /**
- * Return Search Dashboard if connected, otherwise the connection screen.
+ * Return approperiate components.
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/dashboard/wrapped-dashboard.jsx:20]:
```suggestion
 * Return appropriate components.
```

File: projects/packages/search/src/dashboard/utils/get-product-checkout-url.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+/**
+ *
+ * Return the checkout URL for the given product.
+ * It checkes whether the user is connected to Jetpack.
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/utils/get-product-checkout-url.js:4]:
```suggestion
 * It checks whether the user is connected to Jetpack.
```

**Review by bluefuton — COMMENTED** [review]
Tests out well :)

I noticed I was redirected back to `/wp-admin//admin.php?page=jetpack-search` (works but contains extra slash).

Left padding is a bit tight on smaller screen sizes:
Screen Shot 2022-04-20 at 11 59 16
Screen Shot 2022-04-20 at 11 59 05

File: projects/packages/search/src/dashboard/components/dashboard/connection-page.scss (hunk 1 above)
Comment (jsnmoon) [reply projects/packages/search/src/dashboard/components/dashboard/connection-page.scss:9]: That's not possible, I'm afraid. [AdminSectionHero](https://github.com/Automattic/jetpack/blob/master/projects/js-packages/components/components/admin-section/hero/index.jsx#L19) uses obfuscated class names:

Screen Shot 2022-04-21 at 4 29 12 PM

I'll add some code comments pointing this out and warning contributors on this brittle style rule.

File: projects/packages/search/src/dashboard/components/dashboard/search-promotion.scss (hunk 2 above)
Comment (jsnmoon) [reply projects/packages/search/src/dashboard/components/dashboard/search-promotion.scss:9]: Done!

**Comment by jsnmoon** [comment]
> I noticed I was redirected back to `/wp-admin//admin.php?page=jetpack-search` (works but contains extra slash). (@bluefuton)

Addressed all inline feedback, including the extra slash bug, except the following:

> Left padding is a bit tight on smaller screen sizes (@bluefuton)

We'll likely want to tweak the `AdminPage` component for this change. Might I recommend addressing this in a follow-up PR?

**Comment by bluefuton** [comment]
> We'll likely want to tweak the AdminPage component for this change. Might I recommend addressing this in a follow-up PR?

Sounds like a plan. I've added a note to the project board: https://github.com/Automattic/jetpack/projects/34#card-80565581

**Review by bluefuton — APPROVED** [review]
Looks good! 🚢

**Pull request merged**